src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/SplitReservation.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Commands.Reservations.Common;
+using Microsoft.Azure.Management.Reservations.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Reservations;
+using Microsoft.Azure.Commands.Reservations.Models;
+
+namespace Microsoft.Azure.Commands.Reservations.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Split, "AzureRmReservation", DefaultParameterSetName = Constants.ParameterSetNames.CommandParameterSet, SupportsShouldProcess = true), OutputType(typeof(List<PSReservation>))]
+    public class SplitReservation : AzureReservationsCmdletBase
+    {
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet,
+            Mandatory = true)]
+        [ValidateNotNull]
+        public string ReservationOrderId { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet,
+            Mandatory = true)]
+        [ValidateNotNull]
+        public string ReservationId { get; set; }
+
+        [Parameter(Mandatory = true)]
+        [ValidateNotNull]
+        [ValidateCount (2, 2)]
+        public int?[] Quantity { get; set; }
```

Comment:
can any element of this array be null?

Reply:
it cannot be, but in executecmdlet(), when creating split request object, ToList() complained if the type was not nullable.

Reply:
This must be an int

Reply:
Changing code so Quantity is int[], but we still need to pass int?[] to SplitRequest because that is what generated SDK expects.